echo-model/src/main/java/com/netflix/spinnaker/echo/model/Trigger.java

```diff
@@ -86,6 +95,13 @@ public String toString() {
   List<String> expectedArtifactIds;
   Map<String, ?> lastSuccessfulExecution;
 
+  // this is set after deserialization, not in the json representation
+  Pipeline parent;
+
+  public String generateFallbackId() {
+    return UUID.nameUUIDFromBytes(this.toString().getBytes()).toString();
```

Comment:
This is how we generate new ids. Any thoughts on using specific fields vs relying on `toString()`?

Reply:
does seem a bit strange to me ... I've never done it myself.

The `expression` itself combined with parent pipeline id should be sufficiently unique? (I doubt we actually stop someone from registering the same cron expression multiple times tho?)